### ISSUE TITLE
Missing Declaration

### ISSUE THREAD

**Issue by alexandergraul** [QMatrixClient/libqmatrixclient#114]
Hi,
I tried to package this as a shard library and got the following output in my build log. The build succeeded (I circumvented https://github.com/QMatrixClient/libqmatrixclient/issues/113), this is the output that contains the error.

```
'QDebug QMatrixClient::operator<<(QDebug, const QMatrixClient::BaseJob::Status&)' has not been declared within QMatrixClient
[   35s]  QDebug QMatrixClient::operator<<(QDebug dbg, const BaseJob::Status& s)
[   35s]         ^~~~~~~~~~~~~
[   35s] In file included from /home/abuild/rpmbuild/BUILD/libqmatrixclient-0.1/jobs/basejob.cpp:19:0:
[   35s] /home/abuild/rpmbuild/BUILD/libqmatrixclient-0.1/jobs/basejob.h:113:35: note: only here as a friend
[   35s]                      friend QDebug operator<<(QDebug dbg, const Status& s);
[   35s]                                    ^~~~~~~~
```

I can provide a longer log if that helps you.

**Comment by KitsuneRal** [comment]
I'm not sure how the build could succeed with such an error... What compiler are you using?

**Comment by KitsuneRal** [comment]
In general, lookup of friend functions is a mess when it comes to namespaces, so I'm not surprised. Will look into how to better reorganise this.

**Comment by alexandergraul** [comment]
I am using gcc and used make to build it: [link to complete logfile](http://paste.opensuse.org/76580938).

I want to add that I am new to packaging / building from source and while this has built successfully, I could not try it out yet since I am struggling a bit with building Quaternion using the package I built as a dependency (but that has to do with the build service at this point in time)

**Comment by KitsuneRal** [comment]
Ok, that's definitely small and it seems to not even alter anything in the working code - the compiler just whines that it hasn't seen a declaration inside a namespace before the operator definition that implies that namespace. And it seems to be either a fresh addition to GCC, or a consequence of one of those hardening flags (although that would be rather obscure). Letting the issue stay open and wait for a newcomer to valiantly fix it :)

**Comment by KitsuneRal** [comment]
Fixed by 503957c86a84f1be190719a17984df1bb1267658
